session/create_consumer.go

```
@@ -51,16 +49,16 @@ func (consumer *createConsumer) NewRequest() (requestPtr interface{}) {
 // Consume handles requests from endpoint and replies with response
 func (consumer *createConsumer) Consume(requestPtr interface{}) (response interface{}, err error) {
 	request := requestPtr.(*CreateRequest)
-	if consumer.CurrentProposalID != request.ProposalId {
-		return respondWithError(fmt.Sprintf("Proposal doesn't exist: %d", request.ProposalId)), nil
-	}
 
-	sessionInstance, err := consumer.SessionManager.Create(consumer.PeerID)
-	if err != nil {
+	sessionInstance, err := consumer.SessionManager.Create(consumer.PeerID, request.ProposalId)
+	switch err {
+	case nil:
+		return respondWithSession(sessionInstance), nil
+	case ErrorInvalidProposal:
+		return respondWithError(err.Error()), nil
+	default:
```

aren't we hiding error details here?

---

Introduced static error messages